Makefile.pre.in

```diff
@@ -442,25 +442,36 @@ DTRACE_DEPS = \
 all:		@DEF_MAKE_ALL_RULE@
 build_all:	$(BUILDPYTHON) oldsharedmods sharedmods gdbhooks Programs/_testembed python-config
 
-# Compile a binary with profile guided optimization.
-profile-opt:
+# Compile with profile generation
+#
+# Run "make clean" and remove stamp file to force re-build of the profile
+# generation binary.
+profile-gen-stamp:
 	@if [ $(LLVM_PROF_ERR) = yes ]; then \
 		echo "Error: Cannot perform PGO build because llvm-profdata was not found in PATH" ;\
 		echo "Please add it to PATH and run ./configure again" ;\
 		exit 1;\
 	fi
 	@echo "Building with support for profile generation:"
-	$(MAKE) clean
 	$(MAKE) profile-removal
 	$(MAKE) build_all_generate_profile
-	$(MAKE) profile-removal
+	touch $@
+
+# Run task to generate profile information
+#
+# Run "make clean" and remove stamp file to force re-running of profile
+# generation task.
+profile-run-stamp: profile-gen-stamp
 	@echo "Running code to generate profile data (this can take a while):"
+	$(MAKE) profile-removal
 	$(MAKE) run_profile_task
 	$(MAKE) build_all_merge_profile
-	@echo "Rebuilding with profile guided optimizations:"
 	$(MAKE) clean
-	$(MAKE) build_all_use_profile
-	$(MAKE) profile-removal
+	# This is an expensive target to build and it does not have proper
+	# makefile dependancy information.  So, we create a "stamp" file
+	# to record that it has been preformed and does not have to get
```

Comment:
I'd edit this to read

`# to record its completion to avoid rerunning it unnecessarily.`